### ISSUE TITLE
Uniformative assertion error when connecting SQLite database

### ISSUE THREAD

**Issue by Markus28** [optuna/optuna#4375]
### Expected behavior

I tried creating a study with SQLite storage, as described here: https://optuna.readthedocs.io/en/stable/tutorial/20_recipes/001_rdb.html

I get the error message below, which seems to originates from Optuna, directly after a call to Alembic. The following lines are relevant:
https://github.com/optuna/optuna/blob/ead2e13953915bf4ebb50d5880f069a29bb34ffb/optuna/storages/_rdb/storage.py#L1143
https://github.com/sqlalchemy/alembic/blob/08266a49db1ce69224e62fa89f34f03a2f0f9529/alembic/runtime/migration.py#L489

Unfortunately, the message is very uniformative, so I have so far not managed to solve the problem.

### Environment

- Optuna version:3.0.5
- Python version:3.8.14
- OS:Linux-5.15.0-48-generic-x86_64-with-glibc2.35
- Alembic version: 1.9.0

### Error messages, stack traces, or logs

```shell
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/is/sg2/mkrimmel/experiment-management/.venv/lib/python3.8/site-packages/optuna/_convert_positional_args.py", line 63, in converter_wrapper
    return func(**kwargs)
  File "/is/sg2/mkrimmel/experiment-management/.venv/lib/python3.8/site-packages/optuna/study/study.py", line 1147, in create_study
    storage = storages.get_storage(storage)
  File "/is/sg2/mkrimmel/experiment-management/.venv/lib/python3.8/site-packages/optuna/storages/__init__.py", line 32, in get_storage
    return _CachedStorage(RDBStorage(storage))
  File "/is/sg2/mkrimmel/experiment-management/.venv/lib/python3.8/site-packages/optuna/storages/_rdb/storage.py", line 225, in __init__
    self._version_manager.check_table_schema_compatibility()
  File "/is/sg2/mkrimmel/experiment-management/.venv/lib/python3.8/site-packages/optuna/storages/_rdb/storage.py", line 1134, in check_table_schema_compatibility
    current_version = self.get_current_version()
  File "/is/sg2/mkrimmel/experiment-management/.venv/lib/python3.8/site-packages/optuna/storages/_rdb/storage.py", line 1161, in get_current_version
    assert version is not None
AssertionError
```

### Steps to reproduce

```python
>>> import optuna
>>> optuna.create_study(study_name="foo", storage="sqlite:///foo.db")

```

**Comment by Markus28** [comment]
I just happened to find the issue: Downgrading sqlalchemy from 2.0.0 to 1.4.46 solves it! Still, somebody should look into this.

**Comment by nzw0301** [comment]
As you solved, optuna doesn't work with alchemy 2.0. So this issue is duplicated with https://github.com/optuna/optuna/issues/4365.

**Comment by c-bata** [comment]
@Markus28 Could you try `pip install optuna==3.1.0`? It supports SQLAlchemy 2.0 release.
